pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2933,6 +2933,7 @@ protected void interceptCommand(BaseCommand command) throws InterceptException {
 
     @Override
     public void closeProducer(Producer producer) {
+        assert ctx.executor().inEventLoop();
```

Comment:
I don't think this is right fix. there are usecases where adding producer fails and broker wants to close producer in different threads and this skips the closing the producer. using same thread must not be the limitation in this usecase that will cause other issues in async processing as other threads can complete the producer creation/failure. I think right fix will be make `recentlyClosedProducers` thread-safe as it's not being accessed in critical path and won't be the bottleneck.